test/relayHub/RelayHub.test.ts

```diff
@@ -0,0 +1,1199 @@
+
+            it('Should stake only from owner account', async () => {
+                await relayHubInstance.stakeForAddress(relayManager, 1000, {
+                    value: ether('1'),
+                    from: relayOwner
+                });
+                await assert.isRejected(
+                    relayHubInstance.stakeForAddress(relayManager, 1000, {
+                        value: ether('1'),
+                        from: _
+                    }),
+                    'not owner',
+                    'Stake was not made by the owner account'
+                );
+            });
+
+            it('Should NOT be able to unauthorize/unstake a HUB and then perform a new stake with a worker', async () => {
+                await relayHubInstance.stakeForAddress(relayManager, 1000, {
+                    value: ether('1'),
+                    from: relayOwner
+                });
+
+                await relayHubInstance.addRelayWorkers([relayWorker], {
+                    from: relayManager
+                });
+                await relayHubInstance.unlockStake(relayManager, {
+                    from: relayOwner
+                });
+
+                //Verifying stake is now unlocked
+                const stakeInfo = await relayHubInstance.getStakeInfo(
+                    relayManager
+                );
+                const isUnlocked = Number(stakeInfo.withdrawBlock) > 0;
+                assert.isTrue(isUnlocked, 'Stake is not unlocked');
+
+                //Moving blocks to be able to unstake
+                await evmMineMany(Number(stakeInfo.unstakeDelay));
+
+                const gasPrice = toBN('60000000');
+                await assert.isRejected(
+                    relayHubInstance.withdrawStake(relayWorker, {
+                        from: relayOwner,
+                        gasPrice
+                    }),
+                    'not owner',
+                    'Withdraw was made successfully'
+                );
+            });
+
+            it('Should be able to unauthorize/unstake a HUB then stake should be ZERO', async () => {
+                await relayHubInstance.stakeForAddress(relayManager, 1000, {
+                    value: ether('1'),
+                    from: relayOwner
+                });
+
+                await relayHubInstance.addRelayWorkers([relayWorker], {
+                    from: relayManager
+                });
+                await relayHubInstance.unlockStake(relayManager, {
+                    from: relayOwner
+                });
+
+                //Verifying stake is now unlocked
+                let stakeInfo = await relayHubInstance.getStakeInfo(
+                    relayManager
+                );
+                const isUnlocked = Number(stakeInfo.withdrawBlock) > 0;
+                assert.isTrue(isUnlocked, 'Stake is not unlocked');
+
+                //Moving blocks to be able to unstake
+                await evmMineMany(Number(stakeInfo.unstakeDelay));
```

Comment:
We could try to do this async or show a message that would tike a little bit longer.

Reply:
Do you mean we shouldn't wait for the blocks to be mined? If so I don't think we could avoid this. Before allowing the owner to retrieve the stake, they need to wait for a certain number of blocks to be mined, so here we are simulating exactly this.
Please have a look at the [stakeForAddress function](https://github.com/rsksmart/rif-relay-contracts/blob/master/contracts/RelayHub.sol#L312).

Reply:
What I was referring to, its instead of using an async function we could just execute the evmMineMany and keep with the flow. I have seen some tests that do that.

Reply:
Talked with Antonio about this change